src/Solver/Numerov.cpp

```diff
@@ -0,0 +1,249 @@
+        double &pot_2 = potential_values.at(i - 2);
+        double &pot_a = potential_values.at(i);
+
+        double &wave_1 = this->tempWavefunction.at(i - 1);
+        double &wave_2 = this->tempWavefunction.at(i - 2);
+
+        value = 2 * (1.0 - (5 * c) * (energy - pot_1)) * wave_1 - (1.0 + (c) * (energy - pot_2)) * wave_2;
+        value /= (1.0 + (c) * (energy - pot_a));
+    }
+
+}
+
+// A solver of differential equation using Numerov algorithm and selecting non-trivial solutions.
+// It solves the pointed potential using the Numerov algorithm and
+// renormalizing the output wavefunction to 1. To do this it must try the solutions
+// for different energies. The natural solution to the second degree differential equation
+// is the exponential. But my boundary conditions impose 0 at both beginning and end
+// of the wavefunction, so you have to try until you find such solution by finding
+// where the exponential solution changes sign.
+double Numerov::solve(double e_min, double e_max, double e_step)
+{
+    double c, x, first_step, norm, energy = 0.0;
+    int n, sign;
+
+    std::vector< std::vector<double> > potentials;
+    
+    // Get potentials from the main potentials (check if it's separated or not)
+    if (this->potential->isSeparated()) {
+        for (int i = 0; i < this->potential->getSeparatedPotentials().size(); i++) {
+            potentials.push_back(this->potential->getSeparatedPotentials().at(i).getValues());
+        }
+    } 
+    else {
+        potentials.push_back(this->potential->getValues());
+    }
+ 
+    // For each separated potential values...
+    for (int potential_counter = 0; potential_counter < potentials.size(); potential_counter++)
+    {
+        std::vector<double> &this_potential_values = potentials[potential_counter];
+
+        // The final solution energy is the sum of each energy
+        this->solutionEnergy += this->findEnergy(e_min, e_max, e_step, this_potential_values);
+        this->probability     = this->findProbability();
+        this->normalize();
+        this->multiplyWavefunction();
+    }
+    return this->solutionEnergy;
+}
+
+// Applies a bisection algorith to the numerov method to find
+// the energy that gives the non-trivial (non-exponential) solution
+// with the correct boundary conditions
+double Numerov::bisection(double e_min, double e_max, std::vector<double> &this_potential)
+{
+    double energy_middle, fx1, fb, fa;
+    std::cout.precision(17);
+
+    // The number of iterations that the bisection routine needs can be evaluated in advance
+    int itmax = ceil(log2(e_max - e_min) - log2(err)) - 1;
+
+    for (int i = 0; i < itmax; i++)
+    {
+        energy_middle = (e_max + e_min) / 2.0;
+
+        this->findWavefunction(energy_middle, this_potential);
+        fx1 = this->tempWavefunction[this->nbox];
+
+        this->findWavefunction(e_max, this_potential);
+        fb = this->tempWavefunction[this->nbox];
+
+        if (std::abs(fx1) < err)
```

Comment:
@mechanapoleon here you have the infamous err.
Used in this way it generates useless "errors" (which are actually warnings) below.

Reply:
referenced in Issue #17